src/test/java/edu/cornell/kfs/tax/service/impl/TaxProcessingV2ServiceImplTest.java

```diff
@@ -0,0 +1,241 @@
+    private ParameterService mockParameterService;
+    private AtomicReference<TaxBatchConfig> taxConfigHolder;
+
+    @SuppressWarnings("unchecked")
+    @BeforeEach
+    void setUp() throws Exception {
+        taxProcessingV2Service = springContextExtension.getBean(
+                TaxSpringBeans.TAX_PROCESSING_V2_SERVICE, TaxProcessingV2ServiceImpl.class);
+        mockParameterService = springContextExtension.getBean(
+                TaxSpringBeans.BASE_PARAMETER_SERVICE, ParameterService.class);
+        taxConfigHolder = springContextExtension.getBean(
+                TaxSpringBeans.TEST_TAX_CONFIG_HOLDER, AtomicReference.class);
+        taxConfigHolder.set(null);
+    }
+
+    @AfterEach
+    void shutDown() throws Exception {
+        taxConfigHolder = null;
+        mockParameterService = null;
+        taxProcessingV2Service = null;
+    }
+
+
+
+    @SpringXmlTestBeanFactoryMethod
+    public static ParameterService buildMockParameterService() {
+        final ParameterService service = TaxParameterUtils.createUpdatableMockParameterServiceForTaxProcessing();
+        service.createParameter(generate1042SDatesToProcessParameter(KFSConstants.EMPTY_STRING));
+        return service;
+    }
+
+    private static Parameter generate1042SDatesToProcessParameter(final String value) {
+        return TaxParameterUtils.create1042SParameter(
+                CUTaxConstants.TAX_TYPE_1042S + TaxCommonParameterNames.DATES_TO_PROCESS_PARAMETER_SUFFIX, value);
+    }
+
+    @SpringXmlTestBeanFactoryMethod
+    public static AtomicReference<TaxBatchConfig> buildTaxConfigHolder() {
+        return new AtomicReference<>();
+    }
+
+    @SpringXmlTestBeanFactoryMethod
+    public static TaxFileGenerationService buildMockTaxFileGenerationService(
+            final AtomicReference<TaxBatchConfig> configHolder) throws Exception {
+        return new CuMockBuilder<>(TaxFileGenerationService.class)
+                .withAnswer(
+                        service -> service.generateFiles(Mockito.any()),
+                        invocation -> handleTaxFileGenerationServiceCall(invocation, configHolder))
+                .build();
+    }
+
+    private static Object handleTaxFileGenerationServiceCall(
+            final InvocationOnMock invocation, final AtomicReference<TaxBatchConfig> configHolder) {
+        configHolder.set(invocation.getArgument(0));
+        return null;
+    }
+
+    @SpringXmlTestBeanFactoryMethod
+    public static TaxProcessingDao buildMockTaxProcessingDao() {
+        return Mockito.mock(TaxProcessingDao.class);
+    }
+
+    @SpringXmlTestBeanFactoryMethod
+    public static DateTimeService buildSpiedTestDateTimeService() throws Exception {
+        final TestDateTimeServiceImpl dateTimeService = new TestDateTimeServiceImpl();
+        dateTimeService.afterPropertiesSet();
+        return new CuMockBuilder<>(dateTimeService)
+                .withReturn(service -> service.getLocalDateNow(), TEST_DATE_2025_01_10)
```

Comment:
This could be replaced with a lambda method reference 
DateTimeServiceImpl::getLocalDateNow

Reply:
Updating to use method reference.